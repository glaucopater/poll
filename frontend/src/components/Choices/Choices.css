.Choices {
  flex-direction: column;
  align-items: center;
  align-content: center;
}
