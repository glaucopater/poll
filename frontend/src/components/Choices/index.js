import React, { PureComponent } from 'react';
import Choice from '../Choice/';
export default class Choices extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { choices: [] };
  }
  componentDidMount() {
    this.setState({ choices: this.props.choices });
  }

  render() {
    const choices = this.state.choices;
    const { data, voteQuestion } = this.props;
    return (
      <div className="Choices">
        {Object.values(choices).map((p, index) => (
          <Choice key={index} {...p} data={data} voteQuestion={voteQuestion} />
        ))}
      </div>
    );
  }
}
