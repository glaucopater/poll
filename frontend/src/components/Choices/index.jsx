import React, { Component } from 'react';

const style = {
  borderWidth: 1,
  borderColor: '#000',
};
export default class Choices extends Component {
  componentDidMount() {
    console.log(this.props);
  }
  render() {
    return (
      <div className="Choices" style={style}>
        {Object.values(this.props).map((p, index) => (
          <div key={index}>
            <span className="Choice">{p.choice}</span>
            <span className="Votes">{p.votes}</span>
            <span className="Url">{p.url}</span>
          </div>
        ))}
      </div>
    );
  }
}
