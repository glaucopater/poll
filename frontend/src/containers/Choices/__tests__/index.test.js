import { shallow } from 'enzyme';
import React from 'react';
import Choices from '../index';

describe('<Choices/>', () => {
  const props = {
    choices: {},
    voteQuestion: jest.fn(),
  };
  it('renders correctly', () => {
    const wrapper = shallow(<Choices {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
