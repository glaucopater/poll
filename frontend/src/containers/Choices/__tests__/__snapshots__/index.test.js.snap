// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Choices/> renders correctly 1`] = `ShallowWrapper {}`;
